tox.ini

```diff
@@ -45,6 +44,16 @@ deps =
 commands =
     bash -c "./agent/run-unittests"
 
+[testenv:py3-server]
+description = Runs all Python3-based server unit and functional tests
+setenv =
+    PYTHONPATH = {env:PYTHONPATH}:{toxinidir}/server/lib:{toxinidir}/lib
+deps =
+    -r{toxinidir}/test-requirements.txt
+    -r{toxinidir}/server/requirements.txt
+commands =
+    pytest {posargs} ./server/lib/pbench/test/unit
+
```

Comment:
> Have you confirmed that they still work
yes they still work. They are used for running tests in local environment using tox

Reply:
I will check if that will work

Reply:
> If so, should we be running the server unit tests via this mechanism instead of via your changes to jenkins/run-unittests?

This doesn't seems to work

Reply:
I don't know whether it matters, but in the `main` branch, as well as in the `tox` documentation, they don't put a space between the environments specified with the `-e` switch (i.e., to the `jenkins/tox` command in `Pipeline.gy`).

But, in terms of getting the server PyTests to run, it looks like maybe the `PYTHONPATH` isn't being set up correctly.  I see that you _removed_ the settings of it from both the `jenkins/python-setup.sh` (which I don't think matters in that context) and the `tox.ini` which seems like it would cause a problem.  I suggest that you put it back in `tox.ini` and then add a line to the `commands` list to print out its value before invoking PyTest to see if it is set to what we expect.  (The possibility other than it being wrong is that the `configtools` isn't present in `{toxinidir}/lib`, but that seems unlikely...unless `{toxinidir}` has an unexpected value...I suppose you could confirm that at the same time.)

Reply:
```sh 'jenkins/run jenkins/tox -r --current-env -e jenkins-unittests,py3-server'```

jenkins error
```jenkins-unittests: commands succeeded
  py3-server: commands succeeded
  congratulations :)
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/tox/config/__init__.py", line 387, in get
    return self.resolved[name]
KeyError: 'PYTHONPATH'```

Reply:
I also tried removing and adding PYTHONPATH from pbench-setup.sh but it doesn't work that way either.

Reply:
```
setenv =
    PYTHONPATH = {env:PYTHONPATH:}:{toxinidir}/server/lib:{toxinidir}/lib
```

Test cases passed with colon after PYTHONPATH while they are not passing the simple way

Reply:
This is the path list

```
sys.path = [
    '/home/pbench',
    '/home/pbench/server/lib',
    '/home/pbench/lib',
    '/usr/lib64/python39.zip',
    '/usr/lib64/python3.9',
    '/usr/lib64/python3.9/lib-dynload',
    '/usr/lib64/python3.9/site-packages',
    '/usr/lib/python3.9/site-packages',
]
```

Reply:
seems like colon after `PYTHONPATH` which causes an empty string to be used if the environment variable is not defined, worked. Thanks for this point.